src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs

```diff
@@ -291,148 +291,36 @@ private enum RhEHFrameType
 #endif
         }
 
-        // Given an ExceptionID and an address pointing somewhere into a managed module, get
-        // an exception object of a type that the module containing the given address will understand.
-        // This finds the classlib-defined GetRuntimeException function and asks it for the exception object.
-        internal static Exception GetClasslibException(ExceptionIDs id, IntPtr address)
-        {
-#if NATIVEAOT
```

Comment:
This should stay as is. We probably want to keep the GetClasslibException abstraction for the core exception handling as Michal pointed out. This abstraction exists even with INPLACE_RUNTIME today, so you should not need to touch it for INPLACE_RUNTIME.

Also, the [current GetClasslibException callback](https://github.com/dotnet/runtime/blob/e612bf4d0b6f08623092902c34a504e932388664/src/coreclr/nativeaot/System.Private.CoreLib/src/System/RuntimeExceptionHelpers.cs#L41) into CoreLib handles some of the cases differently, and the behavior changes you have introduced by getting rid of it are not good.